### PR TITLE
Generic tree

### PR DESCRIPTION
**Pull request by j-shreya**
Fixes #
Issue number:  #1688 
#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Input is taken dynamically.
- [X] Sample Input / Output is added at the end of file.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- Java
- 

#### Files Added:
- GenericTree_Levelorder_Traversal.java
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by raghav-dalmia** [comment]
are you working on it?

**Comment by j-shreya** [comment]
Yes I am

**Comment by j-shreya** [comment]
Changes requested made. Pls check @SakshiShreya

**Comment by pragyajaiswa05l** [comment]
@jainaman224  @asha15  @raghav-dalmia  please assign it to me in c++.would like to contribute in it as a part of gssoc 2020

**Comment by asha15** [comment]
please squash your commits and let me know

File: GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        System.out.println();
+    }
+
+    public static void main(String[] args) {
+
+        GenericTree_Levelorder_Traversal gt = new GenericTree_Levelorder_Traversal();
+        gt.levelorder();
+    }
+}
+//SAMPLE INPUT: 10 3 20 2 50 0 60 0 30 0 40 0
```
Comment (somya-kapoor) [inline GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java:98]: Leave a line before it

File: GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            // add children of current node to the end of queue
+            for (Node child : rn.children) {
+                queue.addLast(child);
+            }
+        }
+        System.out.println();
+    }
+
+    public static void main(String[] args) {
+
```
Comment (somya-kapoor) [inline GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java:93]: Remove this line

File: GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        System.out.println(str);
+
+        for (Node child : node.children) {
+            display(child);
+        }
+    }
+
+    // level order traversal
+    public void levelorder() {
+
```
Comment (somya-kapoor) [inline GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java:71]: Remove this line

File: GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    public void display() {
+
+        display(root);
+    }
+
+    // function to display generic tree
+    private void display(Node node) {
+
```
Comment (somya-kapoor) [inline GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java:54]: Remove this line

File: GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            Node child = takeInput(nn, i);
+            // add children in array list
+            nn.children.add(child);
+        }
+        return nn;
+    }
+
+    public void display() {
+
```
Comment (somya-kapoor) [inline GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java:48]: Remove this line

File: GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        root = takeInput(null, -1);
+    }
+
+    private Node takeInput(Node parent, int ith) {
+
+        if (parent == null) {
+            System.out.println("Enter the data for Root Node ?");
+        } else {
+            System.out.println(
+                    "Enter data for " + ith + " child of " + parent.data);
```
Comment (somya-kapoor) [inline GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java:26]: 4 space indentation

File: GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    private Node root;
+
+    public GenericTree_Levelorder_Traversal() {
+        root = takeInput(null, -1);
+    }
+
+    private Node takeInput(Node parent, int ith) {
+
```
Comment (somya-kapoor) [inline GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java:21]: Remove this line

File: GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Scanner;
+
+public class GenericTree_Levelorder_Traversal {
+
```
Comment (somya-kapoor) [inline GenericTree_Levelorder_Traversal/GenericTree_Levelorder_Traversal.java:6]: Remove this line

**Comment by j-shreya** [comment]
@somya-kapoor  changes done. Pls check

**Comment by somya-kapoor** [comment]
Are you working on it?

**Pull request merged**